Describe registry use and conformance (#563)
unicode-org/message-format-wg: spec/registry.md

  - Display/edit known message metadata.
  - Restrict input in GUI by providing a dropdown with all viable option values.

## Conformance and Use

_This section is normative._

> To be conformant with MessageFormat 2.0, an implementation MUST implement
> all of the _selectors_ and _functions_ described in the default registry,

Reviewer: Should reword slightly, as _functions_ includes selectors.

```suggestion
To be conformant with MessageFormat 2.0, an implementation MUST implement
all of the formatting and selection _functions_ described in the default registry,
```

including all of the _options_ and _option_ values, _operands_ and outputs
described by the default registry.

Implementations are not required to provide a registry nor to read or interpret
a copy of this registry in order to be conformant.

> The MessageFormat 2.0 Registry was created to describe the core set of _selectors_
> and _functions_, including _operands_, _options_, _option_ values, as well
> as expected outputs.

Reviewer: Actually, the registry does not (at least currently) define expected outputs.

```suggestion
The MessageFormat 2.0 Registry was created to describe
the core set of formatting and selection _functions_,
including _operands_, _options_, and _option_ values.
```

The descriptions in this registry are intended to describe the specific required

> signatures for _selectors_ and _functions_ that all implementations of MessageFormat 2.0

Reviewer: ```suggestion
signatures for _functions_ that all implementations of MessageFormat 2.0
```

MUST implement.

> The goal is to ensure message interoperability between implementations,
> regardless of programming language or runtime environment.

Reviewer: I don't think we can _ensure_ interop, but we can at least try.

```suggestion
The goal is to enable message interoperability between implementations,
regardless of programming language or runtime environment.
```

> This ensures that developers do not have to relearn core MessageFormat syntax
> when moving between platforms

Reviewer: ```suggestion
This ensures that developers do not have to relearn core MessageFormat syntax
and functionality when moving between platforms
```

and that translators do not need to know about the runtime environment for most
selection or formatting operations.

> The registry provides a machine-readable description of _selectors_ and _functions_

Reviewer: ```suggestion
The registry provides a machine-readable description of _functions_
```
